packages/block-editor/src/components/block-list/use-block-props/use-selected-block-event-handlers.js

```diff
@@ -76,7 +76,34 @@ export function useEventHandlers( { clientId, isSelected } ) {
 			 * @param {DragEvent} event Drag event.
 			 */
 			function onDragStart( event ) {
-				event.preventDefault();
+				const { ownerDocument } = node;
+				const { defaultView } = ownerDocument;
+				const selection = defaultView.getSelection();
+				let { anchorNode, focusNode } = selection;
+				if ( anchorNode !== anchorNode.ELEMENT_NODE ) {
+					anchorNode = anchorNode.parentElement;
+				}
+				if ( focusNode !== focusNode.ELEMENT_NODE ) {
+					focusNode = focusNode.parentElement;
+				}
+				if (
+					node !== event.target ||
+					( node.contains( anchorNode ) &&
+						anchorNode.isContentEditable ) ||
+					( node.contains( focusNode ) &&
+						focusNode.isContentEditable )
+				) {
+					event.preventDefault();
+					return;
+				}
```

Comment:
Would you mind adding some comments here to clarify what the code does, I think you try to guess the cases where you want to bail early and disable the bahavior but it would be cool to explain the cases.

Reply:
This has changed now.